src/engine/controls/cuecontrol.cpp

```diff
@@ -22,19 +23,39 @@ static const double CUE_MODE_NUMARK = 3.0;
 static const double CUE_MODE_MIXXX_NO_BLINK = 4.0;
 static const double CUE_MODE_CUP = 5.0;
 
+constexpr double kNoColorControlValue = -1;
```

Comment:
I think we wanted to get rid of this? This triggers not well designed code paths everywhere. 
Can we use the default color instead?

Reply:
No, this is used for `HotcueControl` (not `Cue`). We need this because a `HotcueControl` is not always associated with a `Cue` - this is the case if a hotcue is not set or no track is loaded. I don't want my performance pads to light up in the default color in these cases.

Reply:
Yes, but in any case the control value should be `-1`. We do the same for hotcue positions, etc.

Reply:
We can consider it, but there should not be a need at all. Setting the cue position to -1 means it is not set, this is a programmatic requirement and should be the only one. 
Currently parts of the code are looking to the position others to the color. That is the conceptual issue that should be fixed.  
Once this is the case, the color can remain where it was. This has the benefit that we can assert that the hotcue_nn_color CO alwasy holds a valid color.  So there is no need for extended handling.

Reply:
> Once this is the case, the color can remain where it was. This has the benefit that we can assert that the hotcue_nn_color CO alwasy holds a valid color. So there is no need for extended handling.

I can't assert that `hotcue_nn_color` always holds a valid color, because COs could be set to any value. What if I start mixxx with `--developer` and set the CO via the "Developer Tools" to `-12.74`? We need the value checks anyway.

Reply:
I don't mind if we trigger a debug assert in this case. If you are, we can register the change request hook and reject all invalid values.

Reply:
Ok, I will issue that in my follow up PR.